### DISCUSSION TITLE
Maybe a little too easy? - How to edit existing connections or add custom properties to connections?

### DISCUSSION THREAD

**Discussion by Ooonsta** [wg-easy/wg-easy#1158] [Ideas]
I couldn't for the life of me work out how to _edit_ a connection. You can only really add new connections via the GUI? I went down the path of mapping a volume outside of docker to try and edit the wg0.conf file but had no success. When I started looking at docker custom entry points - 

```
#!/bin/sh
# Custom entrypoint script to prevent configuration overwrite

# Check if wg0.conf exists
if [ ! -f /etc/wireguard/wg0.conf ]; then
    # If not, copy the default config (optional)
    cp /default-config/wg0.conf /etc/wireguard/wg0.conf
fi

# Start WG-Easy
exec /entrypoint.sh "$@"
```

 ... I thought there might be an easier way. Is there a way of adding custom properties to connections?

Thanks in advance.

**Comment by eandersons** [comment]
What exactly do you want to edit in a connection?

Your changes in `wg0.conf` were overriden because `wg0.conf` is recreated from `wg0.json`, if I recall correctly.
You could try to edit `wg0.json` to see if that helps you achieve the desired outcome.

**Reply by Ooonsta** [reply]
Thanks very much for that. I gave it a crack (editing the .json file) and still couldn't get it going. After I removed and re-did docker the .json file retained my settings but these were not carried over to wg-easy so I probably screwed up some syntax in there...

I wanted (expected) to be able add details to the existing connection profile or have the ability when you create a new connection to specify variables then and there via the GUI, like -

[Interface]  (is this the same as "server" in the .json file?)
PrivateKey = 
Address = 
MTU = 

[Peer]
PublicKey = 
AllowedIPs =
Endpoint = 
etc...

**Reply by eandersons** [reply]
See the method `__saveConfig` for values that are written to `wg0.conf` from `wg0.json`.

https://github.com/wg-easy/wg-easy/blob/94d87681c384689cb4aa371d9d9c3a924f40ce5f/src/lib/WireGuard.js#L89-L125

If you want control over more WireGuard configuration values, you will have to modify the source.
